sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -107,6 +109,31 @@ public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
             .doOnError(throwable -> logger.warning("<-- HTTP FAILED: ", throwable));
     }
 
+    @Override
+    public HttpResponse processSynchronously(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // No logging will be performed, trigger a no-op.
+        if (httpLogDetailLevel == HttpLogDetailLevel.NONE) {
+            return next.processSynchronously();
+        }
+
+        final ClientLogger logger = new ClientLogger((String) context.getData("caller-method").orElse(""));
+        final long startNs = System.nanoTime();
+
+        requestLogger.logRequestSynchronously(logger, getRequestLoggingOptions(context));
+        try {
+            HttpResponse response = next.processSynchronously();
+            if (response != null) {
+                response = responseLogger.logResponseSynchronously(
+                    logger, getResponseLoggingOptions(response, startNs, context));
+            }
+            return response;
+        } catch (RuntimeException e) {
+            logger.warning("<-- HTTP FAILED: ", e);
+            // TODO (kasobol-msft) this is probably overkill. suppression needed.
+            throw logger.logExceptionAsWarning(e);
```

Comment:
Yeah, let's suppress this needing to be logged as it's double logging the exception

Reply:
I'll keep the todo for now.